evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -353,86 +362,93 @@ def test_simple_imputer_with_none():
     assert_frame_equal(expected, transformed, check_dtype=False)
 
 
-def test_simple_imputer_supports_natural_language_constant():
+@pytest.mark.parametrize("na_type", ["python_none", "numpy_nan", "pandas_na"])
+@pytest.mark.parametrize("data_type", ["Categorical", "NaturalLanguage"])
+def test_simple_imputer_supports_natural_language_and_categorical_constant(
+    na_type, data_type
+):
+    na_type = {"python_none": None, "numpy_nan": np.nan, "pandas_na": pandas.NA}[
+        na_type
+    ]
     X = pd.DataFrame(
         {
-            "cat with None": ["a", "b", "a", None],
-            "natural language col": ["free-form text", "will", "be imputed", None],
+            "Categorical": ["a", "b", "a", na_type],
+            "NaturalLanguage": ["free-form text", "will", "be imputed", na_type],
         }
     )
+    X = X[[data_type]]
     y = pd.Series([0, 0, 1, 0, 1])
-    X.ww.init(
-        logical_types={
-            "cat with None": "categorical",
-            "natural language col": "NaturalLanguage",
-        }
-    )
+
+    X.ww.init(logical_types={data_type: data_type})
     imputer = SimpleImputer(impute_strategy="constant", fill_value="placeholder")
     imputer.fit(X, y)
     transformed = imputer.transform(X, y)
     expected = pd.DataFrame(
         {
-            "cat with None": pd.Series(
-                ["a", "b", "a", "placeholder"], dtype="category"
-            ),
-            "natural language col": pd.Series(
-                ["free-form text", "will", "be imputed", "placeholder"], dtype="string"
+            "Categorical": pd.Series(["a", "b", "a", "placeholder"], dtype="category"),
+            "NaturalLanguage": pd.Series(
+                ["free-form text", "will", "be imputed", pd.NA], dtype="string"
```

Comment:
Might be a little redundant, but the imputer now ignores the natural language columns.

Reply:
I think it only ignores natural language columns if the only column in the dataframe is a natural language column?

```python
from evalml.pipelines.components import SimpleImputer
import pandas as pd
import woodwork as ww
import pytest

X = pd.DataFrame({"a": ["foo", "bar", "baz"], "b": [1, 2, 3]})
X.ww.init(logical_types={"a": "NaturalLanguage"})
si = SimpleImputer()

with pytest.raises(ValueError):
    si.fit_transform(X)
```

I think it might be better to raise an error if non-imputable types are present?

Reply:
@freddyaboulton I'm more of a fan of ignoring the columns if our imputer explicitly (we should not this in the docstring if we pursue this path) does not support natural language columns and that's the design decision I made with all of the select column transformers. If we raise an error it would mean that we would error out during search if the input contains a natural language column with nulls (but I'm unsure how the `NaturalLanguageFeaturizer` deals with nulls. Maybe raising a warning is a happy medium? 

@chukarsten if we want to keep with the ignoring path - we would need to add the natural language columns back in after `transform`-ing as well.